hudi-spark/src/test/scala/org/apache/hudi/functional/TestCOWDataSource.scala

```diff
@@ -194,4 +195,42 @@ class TestCOWDataSource extends HoodieClientTestBase {
       .load(basePath)
     assertEquals(hoodieIncViewDF2.count(), insert2NewKeyCnt)
   }
+
+  @Test def testCopyOnWriteStorageWithMetadata() {
```

Comment:
should we test via `TestHoodieROTablePathFilter` directly?  in the future, Datasource may not call the path filter at all for e.g?

Reply:
any way to just parameterize this entire test with and without metadata. might be easier to extend down the line as well?

Reply:
Made the test parameterized.

Reply:
Testing from `TestHoodieROTablePathFilter` directly seems to be quite a task in itself, given the logic inside it. `HoodieTestTable` will have to be modified to be able to simulate metadata table, and create relevant HFiles inside the metadata table. Let me create a Jira for this for now and come to it later.

For now want to merge this, and focus on `get all partition paths` before I head for my time off.